libs/ColorKit/source/ColorKit.cpp

```diff
@@ -1,15 +1,50 @@
 // Leka - LekaOS
 // Copyright 2022 APF France handicap
 // SPDX-License-Identifier: Apache-2.0
+#pragma once
 
 #include "ColorKit.h"
 #include <cmath>
 
+#include "ColorPosition.h"
+#include "Utils.h"
+
 namespace leka {
 
 auto ColorKit::colorGradient(const RGB &color1, const RGB &color2, const float &position) -> RGB
 {
 	return oklab2rgb(lerp(ColorKit::internal::rgb2oklab(color1), ColorKit::internal::rgb2oklab(color2), position));
 }
 
+auto ColorKit::create255RGBGradient(std::vector<ColorPosition> input_colors, const RGB &color_start,
+									const RGB &color_end) -> std::array<RGB, 255>
+{
+	std::array<RGB, 255> return_colors;
+	auto size		 = static_cast<uint8_t>(input_colors.size());
+	auto first_input = input_colors[0];
+
+	auto last_input = input_colors[size - 1];
+
+	for (uint8_t index(0); index < first_input.position; index++) {
+		float index_color	 = leka::utils::math::map(index, uint8_t {0}, uint8_t {first_input.position}, 0.F, 1.F);
+		return_colors[index] = ColorKit::colorGradient(color_start, first_input.color, index_color);
+	}
+
+	for (uint8_t i(0); i < size - 1; i++) {
+		for (uint8_t index(input_colors[i].position); index < input_colors[i + 1].position; index++) {
```

Comment:
si c'est des for loop les un dans les autres, il faut utiliser `i, j, k` etc.

```suggestion
		for (auto j = input_colors[i].position; j < input_colors[i + 1].position; ++j) {
```